### DISCUSSION TITLE
What's the thought behind using platform independent implementations on Android?

### DISCUSSION THREAD

**Discussion by FooIbar** [pdvrieze/xmlutil#221] [General]
https://github.com/pdvrieze/xmlutil/blob/0c9e14016b9cfc1682642a8d6ad40ae64cf6ef6f/core/android/src/main/kotlin/nl/adaptivity/xmlutil/core/AndroidStreamingFactory.kt#L61-L66
Isn't the purpose of using Android artifacts to use platform implementations?
Also `KtXmlReader` won't throw `XmlPullParserException`.

**Comment by pdvrieze** [comment]
Basically the generic implementation is an adaptation of the (BSD licensed) code from Android, but with fixes/enhancements (like emoji support). I will fix it to use the `AndroidXmlReader` by default (but remove it as included by default for the serialization module). For writing XML the library has never used the native xml writer.

The exception catching is an oversight.

**Reply by FooIbar** [reply]
> For writing XML the library has never used the native xml writer.

It's somewhat confusing that `newWriter` and `newGenericWriter` have the same implementation on Android, making it inconsistent with other platforms.

**Reply by pdvrieze** [reply]
It is no longer the case for Android. Note that for native, generic and "regular" is still the same implementation. Note also that the core library now by default uses the generic implementation unless you add the core-android/core-jdk packages (for the jdk this is automatically done by serialization).

**Reply by FooIbar** [reply]
Actually, does `serialization-android` really need to depend on `core-android` if the generic implementation is just superior?

**Reply by pdvrieze** [reply]
Serialization-android will not as adding it will change the actual behaviour. And in case of Android, the implementation is fairly minimal, so it indeed mainly exists for legacy reasons (or if someone has an `XmlPullParser` they want to plug into the serialization framework.
